gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityServiceEntityFetcher.java

```diff
@@ -57,7 +60,18 @@ private EntityQueryRequest buildRequest(
 
     addGroupBys(exploreRequest, builder);
     addSelections(requestContext, exploreRequest, builder);
+
     builder.setLimit(DEFAULT_ENTITY_REQUEST_LIMIT);
+
+    // TODO: Push group by down to EQS
+    // If there is a group by, specify a large limit and track actual limit, offset and order by
+    // expression list, so we can compute these once the we get the results.
+    if (requestContext.hasGroupBy()) {
+      // Will need to do the ordering, limit and offset ourselves after we get the group by results
```

Comment:
We aren't pushing order by to EDS in case of explore. EDS handler is only called when 
- there is a group by, order by and selections on EDS. In this case, we handle it in memory. We can push group by to EDS along with order by, which is being called out by the TODO comment. That requires a separate effort
- there is an aggregated selection on EDS (like DISTINCT COUNT of APIs). There is no order by here. And, the limit in this case would be 1 inherently

Reply:
Second case makes sense.

First case - I thought we _were_ pushing down group by (line 61). Discussed offline, but since this is the final query and isn't being joined with anything in memory (right?) we should be able to push everything down unless there's something missing support in EQS.

Reply:
Tried out different queries. We don't support ordering in document store on function expressions. Since, that's the case, we can't push limit/offset and order bys to EQS

We are already pushing group bys to EQS. Pushing limit and offset to EQS, only if there are no order by expressions (which shouldn't ideally be the case, since group by mostly comes with order by)

```
orderBy: [
  {
    aggregation: COUNT
    size: null
    direction: DESC
    keyExpression: { key: "name" }
  }
]
```